iotwifi/iotwifi.go

```diff
@@ -110,23 +110,29 @@ func RunWifi(log bunyan.Logger, messages chan CmdMessage, cfgLocation string) {
 
 	time.Sleep(10 * time.Second)
 
+	// Start supplicant and attempt to connect
 	command.StartWpaSupplicant()
 
-	// Scan
+	// Do a single scan
 	time.Sleep(5 * time.Second)
 	wpacfg.ScanNetworks()
 
 	command.StartDnsmasq()
 
-	// TODO: check to see if we are stuck in a scanning state before
-	// if in a scanning state set a timeout before resetting
+	// monitor for a future connection - shut down AP when it occurs
 	go func() {
 		for {
-			wpacfg.ScanNetworks()
 			time.Sleep(30 * time.Second)
+			if status, ok := wpacfg.Status(); ok == nil && status["wpa_state"] == "COMPLETED" {
```

Comment:
Should `wpacfg.Status()` be using the `APStatus()`function that was added below?

Additionally returning `nil` in an `ok` introduces a 3 value boolean. Could we return false instead in the `Status()` implementation?

Reply:
.Status is correct in this case since we are looking at the wlan connection (AP is for the AP).  It probably could be modified to .WLANStatus but I didn't want to introduce breaking changes on the endpoints.

Reply:
Good catch on the return - converting ok to err